### ISSUE TITLE
qemu is terminated after boot command is input when no communicator is used

### ISSUE THREAD

**Issue by ltrager** [hashicorp/packer#10167]
#### Overview of the Issue

Packer 1.6.3 and above exits after the boot command is input. No output/disk image is generated. Packer seems to think the install completed successfully and tries to run post-processors.

It looks like because no communicator is configured Packer does not wait for the install to finish and just kills qemu. While my template sets `communicator: none` `shutdown_timeout` is configured.

```
2020/10/23 16:40:44 packer-builder-qemu plugin: Special code '<enter>' found, replacing with: 0xFF0D
    qemu: No communicator is configured -- skipping StepWaitGuestAddress
==> qemu: Halting the virtual machine...
2020/10/23 16:40:45 packer-builder-qemu plugin: [INFO] communicator disabled, will not connect
2020/10/23 16:40:45 packer-builder-qemu plugin: Unable to load communicator config from state to populate provisionHookData
2020/10/23 16:40:45 packer-builder-qemu plugin: Running the provision hook
2020/10/23 16:40:45 packer-builder-qemu plugin: VM shut down.
```

I initially reported this in #10135 which also showed a crash. The crash has been fixed but qemu is still killed before an image is generated. @SwampDragons asked I open this if still seeing a problem.

#### Reproduction Steps

1. Download/build Packer 1.6.3+, the log above is from master(25f4e2477)
2. `git clone --recurse-submodules git@github.com:canonical/packer-maas.git`
3. `cd packer-maas/centos8`
4. `sudo PACKER_LOG=1 packer build centos8.json`

### Packer version

Packer v1.6.5-dev (25f4e2477)

### Simplified Packer Buildfile

https://github.com/canonical/packer-maas/tree/master/centos8

### Operating system and Environment details

OS: Ubuntu 20.10 Groovy Gorilla
Arch: x86_64
Kernel: 5.8.0-25-generic
qemu: 1:5.0-5ubuntu9

### Log Fragments and crash.log files

No crash.log is generated. Debug output below

```
$ sudo PACKER_LOG=1 /home/lee/go/bin/packer build centos8.json
2020/10/23 16:40:23 [INFO] Packer version: 1.6.5-dev (25f4e2477) [go1.15.3 linux amd64]
2020/10/23 16:40:23 Checking 'PACKER_CONFIG' for a config file path
2020/10/23 16:40:23 'PACKER_CONFIG' not set; checking the default config file path
2020/10/23 16:40:23 Attempting to open config file: /root/.packerconfig
2020/10/23 16:40:23 [WARN] Config file doesn't exist: /root/.packerconfig
2020/10/23 16:40:23 Setting cache directory: /home/lee/packer-maas/centos8/packer_cache
2020/10/23 16:40:23 Creating plugin client for path: /home/lee/go/bin/packer
2020/10/23 16:40:23 Starting plugin: /home/lee/go/bin/packer []string{"/home/lee/go/bin/packer", "plugin", "packer-builder-qemu"}
2020/10/23 16:40:23 Waiting for RPC address for: /home/lee/go/bin/packer
2020/10/23 16:40:23 packer-builder-qemu plugin: [INFO] Packer version: 1.6.5-dev (25f4e2477) [go1.15.3 linux amd64]
2020/10/23 16:40:23 packer-builder-qemu plugin: Checking 'PACKER_CONFIG' for a config file path
2020/10/23 16:40:23 packer-builder-qemu plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/10/23 16:40:23 packer-builder-qemu plugin: Attempting to open config file: /root/.packerconfig
2020/10/23 16:40:23 packer-builder-qemu plugin: [WARN] Config file doesn't exist: /root/.packerconfig
2020/10/23 16:40:23 packer-builder-qemu plugin: Setting cache directory: /home/lee/packer-maas/centos8/packer_cache
2020/10/23 16:40:23 packer-builder-qemu plugin: args: []string{"packer-builder-qemu"}
2020/10/23 16:40:23 packer-builder-qemu plugin: Plugin address: unix /tmp/packer-plugin064439128
2020/10/23 16:40:23 packer-builder-qemu plugin: Waiting for connection...
2020/10/23 16:40:23 Received unix RPC address for /home/lee/go/bin/packer: addr is /tmp/packer-plugin064439128
2020/10/23 16:40:23 packer-builder-qemu plugin: Serving a plugin connection...
2020/10/23 16:40:23 Creating plugin client for path: /home/lee/go/bin/packer
2020/10/23 16:40:23 Starting plugin: /home/lee/go/bin/packer []string{"/home/lee/go/bin/packer", "plugin", "packer-post-processor-shell-local"}
2020/10/23 16:40:23 Waiting for RPC address for: /home/lee/go/bin/packer
2020/10/23 16:40:23 packer-post-processor-shell-local plugin: [INFO] Packer version: 1.6.5-dev (25f4e2477) [go1.15.3 linux amd64]
2020/10/23 16:40:23 packer-post-processor-shell-local plugin: Checking 'PACKER_CONFIG' for a config file path
2020/10/23 16:40:23 packer-post-processor-shell-local plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/10/23 16:40:23 packer-post-processor-shell-local plugin: Attempting to open config file: /root/.packerconfig
2020/10/23 16:40:23 packer-post-processor-shell-local plugin: [WARN] Config file doesn't exist: /root/.packerconfig
2020/10/23 16:40:23 packer-post-processor-shell-local plugin: Setting cache directory: /home/lee/packer-maas/centos8/packer_cache
2020/10/23 16:40:23 packer-post-processor-shell-local plugin: args: []string{"packer-post-processor-shell-local"}
2020/10/23 16:40:23 packer-post-processor-shell-local plugin: Plugin address: unix /tmp/packer-plugin511924660
2020/10/23 16:40:23 packer-post-processor-shell-local plugin: Waiting for connection...
2020/10/23 16:40:23 Received unix RPC address for /home/lee/go/bin/packer: addr is /tmp/packer-plugin511924660
2020/10/23 16:40:23 packer-post-processor-shell-local plugin: Serving a plugin connection...
2020/10/23 16:40:23 Preparing build: qemu
2020/10/23 16:40:23 packer-builder-qemu plugin: use detected accelerator: kvm
2020/10/23 16:40:23 packer-builder-qemu plugin: CpuCount 0 too small, using default: 1
2020/10/23 16:40:23 Build debug mode: false
2020/10/23 16:40:23 Force build: false
2020/10/23 16:40:23 On error: 
qemu: output will be in this color.
2020/10/23 16:40:23 Waiting on builds to complete...
2020/10/23 16:40:23 Starting build run: qemu

2020/10/23 16:40:23 Running builder: qemu
2020/10/23 16:40:23 [INFO] (telemetry) Starting builder qemu
2020/10/23 16:40:23 packer-builder-qemu plugin: Qemu path: /usr/bin/qemu-system-x86_64, Qemu Image page: /usr/bin/qemu-img
==> qemu: Retrieving ISO
2020/10/23 16:40:23 packer-builder-qemu plugin: Acquiring lock for: https://mirrors.edge.kernel.org/centos/8.2.2004/isos/x86_64/CentOS-8.2.2004-x86_64-boot.iso?checksum=sha256%3Ac67876a5602faa17f68b40ccf2628799b87454aa67700f0f57eec15c6ccdd98c (/home/lee/packer-maas/centos8/packer_cache/8d71e8377b300414b6e2ac9d20cf8724f37070f7.iso.lock)
==> qemu: Trying https://mirrors.edge.kernel.org/centos/8.2.2004/isos/x86_64/CentOS-8.2.2004-x86_64-boot.iso
==> qemu: Trying https://mirrors.edge.kernel.org/centos/8.2.2004/isos/x86_64/CentOS-8.2.2004-x86_64-boot.iso?checksum=sha256%3Ac67876a5602faa17f68b40ccf2628799b87454aa67700f0f57eec15c6ccdd98c
2020/10/23 16:40:25 packer-builder-qemu plugin: Leaving retrieve loop for ISO
2020/10/23 16:40:25 packer-builder-qemu plugin: No floppy files specified. Floppy disk will not be made.
2020/10/23 16:40:25 packer-builder-qemu plugin: No CD files specified. CD disk will not be made.
2020/10/23 16:40:25 packer-builder-qemu plugin: [INFO] Creating disk with Path: output-qemu/packer-qemu and Size: 4G
2020/10/23 16:40:25 packer-builder-qemu plugin: Executing qemu-img: []string{"create", "-f", "qcow2", "output-qemu/packer-qemu", "4G"}
==> qemu: https://mirrors.edge.kernel.org/centos/8.2.2004/isos/x86_64/CentOS-8.2.2004-x86_64-boot.iso?checksum=sha256%3Ac67876a5602faa17f68b40ccf2628799b87454aa67700f0f57eec15c6ccdd98c => /home/lee/packer-maas/centos8/packer_cache/8d71e8377b300414b6e2ac9d20cf8724f37070f7.iso
==> qemu: Creating required virtual machine disks
2020/10/23 16:40:25 packer-builder-qemu plugin: stdout: Formatting 'output-qemu/packer-qemu', fmt=qcow2 size=4294967296 cluster_size=65536 lazy_refcounts=off refcount_bits=16
2020/10/23 16:40:25 packer-builder-qemu plugin: stderr:
2020/10/23 16:40:25 packer-builder-qemu plugin: Found available port: 8621 on IP: 0.0.0.0
==> qemu: Starting HTTP server on port 8621
    qemu: No communicator is set; skipping port forwarding setup.
==> qemu: Looking for available port between 5900 and 6000 on 127.0.0.1
2020/10/23 16:40:25 packer-builder-qemu plugin: Looking for available port between 5900 and 6000 on 127.0.0.1
2020/10/23 16:40:25 packer-builder-qemu plugin: Found available port: 5914 on IP: 127.0.0.1
2020/10/23 16:40:25 packer-builder-qemu plugin: Found available VNC port: 5914 on IP: 127.0.0.1
2020/10/23 16:40:25 packer-builder-qemu plugin: Qemu --version output: QEMU emulator version 5.0.0 (Debian 1:5.0-5ubuntu9)
2020/10/23 16:40:25 packer-builder-qemu plugin: Copyright (c) 2003-2020 Fabrice Bellard and the QEMU Project developers
2020/10/23 16:40:25 packer-builder-qemu plugin: Qemu version: 5.0.0
==> qemu: Starting VM, booting from CD-ROM
    qemu: view the screen of the VM, connect via VNC without a password to
    qemu: vnc://127.0.0.1:14
    qemu: The VM will be run headless, without a GUI. If you want to
    qemu: view the screen of the VM, connect via VNC without a password to
2020/10/23 16:40:25 packer-builder-qemu plugin: Qemu Builder has no floppy files, not attaching a floppy.
    qemu: vnc://127.0.0.1:14
==> qemu: Overriding default Qemu arguments with qemuargs template option...
2020/10/23 16:40:25 packer-builder-qemu plugin: Executing /usr/bin/qemu-system-x86_64: []string{"-serial", "stdio", "-name", "packer-qemu", "-machine", "type=pc,accel=kvm", "-netdev", "user,id=user.0", "-vnc", "127.0.0.1:14", "-device", "virtio-net,netdev=user.0", "-boot", "once=d", "-m", "2048M", "-drive", "file=output-qemu/packer-qemu,if=virtio,cache=writeback,discard=ignore,format=qcow2", "-drive", "file=/home/lee/packer-maas/centos8/packer_cache/8d71e8377b300414b6e2ac9d20cf8724f37070f7.iso,index=0,media=cdrom"}
2020/10/23 16:40:25 packer-builder-qemu plugin: Started Qemu. Pid: 31939
2020/10/23 16:40:25 packer-builder-qemu plugin: Qemu stderr: qemu-system-x86_64: warning: host doesn't support requested feature: CPUID.80000001H:ECX.svm [bit 2]
==> qemu: Waiting 3s for boot...
==> qemu: Connecting to VM via VNC (127.0.0.1:5914)
2020/10/23 16:40:30 packer-builder-qemu plugin: Connected to VNC desktop: QEMU (packer-qemu)
==> qemu: Typing the boot command over VNC...
2020/10/23 16:40:30 packer-builder-qemu plugin: Special code '<up>' found, replacing with: 0xFF52
2020/10/23 16:40:30 packer-builder-qemu plugin: Special code '<tab>' found, replacing with: 0xFF09
2020/10/23 16:40:30 packer-builder-qemu plugin: Sending char ' ', code 0x20, shift false
2020/10/23 16:40:30 packer-builder-qemu plugin: Sending char 'i', code 0x69, shift false
2020/10/23 16:40:31 packer-builder-qemu plugin: Sending char 'n', code 0x6E, shift false
2020/10/23 16:40:31 packer-builder-qemu plugin: Sending char 's', code 0x73, shift false
2020/10/23 16:40:31 packer-builder-qemu plugin: Sending char 't', code 0x74, shift false
2020/10/23 16:40:31 packer-builder-qemu plugin: Sending char '.', code 0x2E, shift false
2020/10/23 16:40:31 packer-builder-qemu plugin: Sending char 'k', code 0x6B, shift false
2020/10/23 16:40:32 packer-builder-qemu plugin: Sending char 's', code 0x73, shift false
2020/10/23 16:40:32 packer-builder-qemu plugin: Sending char '=', code 0x3D, shift false
2020/10/23 16:40:32 packer-builder-qemu plugin: Sending char 'h', code 0x68, shift false
2020/10/23 16:40:32 packer-builder-qemu plugin: Sending char 't', code 0x74, shift false
2020/10/23 16:40:32 packer-builder-qemu plugin: Sending char 't', code 0x74, shift false
2020/10/23 16:40:33 packer-builder-qemu plugin: Sending char 'p', code 0x70, shift false
2020/10/23 16:40:33 packer-builder-qemu plugin: Sending char ':', code 0x3A, shift true
2020/10/23 16:40:33 packer-builder-qemu plugin: Sending char '/', code 0x2F, shift false
2020/10/23 16:40:33 packer-builder-qemu plugin: Sending char '/', code 0x2F, shift false
2020/10/23 16:40:34 packer-builder-qemu plugin: Sending char '1', code 0x31, shift false
2020/10/23 16:40:34 packer-builder-qemu plugin: Sending char '0', code 0x30, shift false
2020/10/23 16:40:34 packer-builder-qemu plugin: Sending char '.', code 0x2E, shift false
2020/10/23 16:40:34 packer-builder-qemu plugin: Sending char '0', code 0x30, shift false
2020/10/23 16:40:34 packer-builder-qemu plugin: Sending char '.', code 0x2E, shift false
2020/10/23 16:40:35 packer-builder-qemu plugin: Sending char '2', code 0x32, shift false
2020/10/23 16:40:35 packer-builder-qemu plugin: Sending char '.', code 0x2E, shift false
2020/10/23 16:40:35 packer-builder-qemu plugin: Sending char '2', code 0x32, shift false
2020/10/23 16:40:35 packer-builder-qemu plugin: Sending char ':', code 0x3A, shift true
2020/10/23 16:40:36 packer-builder-qemu plugin: Sending char '8', code 0x38, shift false
2020/10/23 16:40:36 packer-builder-qemu plugin: Sending char '6', code 0x36, shift false
2020/10/23 16:40:36 packer-builder-qemu plugin: Sending char '2', code 0x32, shift false
2020/10/23 16:40:36 packer-builder-qemu plugin: Sending char '1', code 0x31, shift false
2020/10/23 16:40:37 packer-builder-qemu plugin: Sending char '/', code 0x2F, shift false
2020/10/23 16:40:37 packer-builder-qemu plugin: Sending char 'c', code 0x63, shift false
2020/10/23 16:40:37 packer-builder-qemu plugin: Sending char 'e', code 0x65, shift false
2020/10/23 16:40:37 packer-builder-qemu plugin: Sending char 'n', code 0x6E, shift false
2020/10/23 16:40:37 packer-builder-qemu plugin: Sending char 't', code 0x74, shift false
2020/10/23 16:40:38 packer-builder-qemu plugin: Sending char 'o', code 0x6F, shift false
2020/10/23 16:40:38 packer-builder-qemu plugin: Sending char 's', code 0x73, shift false
2020/10/23 16:40:38 packer-builder-qemu plugin: Sending char '8', code 0x38, shift false
2020/10/23 16:40:38 packer-builder-qemu plugin: Sending char '.', code 0x2E, shift false
2020/10/23 16:40:38 packer-builder-qemu plugin: Sending char 'k', code 0x6B, shift false
2020/10/23 16:40:39 packer-builder-qemu plugin: Sending char 's', code 0x73, shift false
2020/10/23 16:40:39 packer-builder-qemu plugin: Sending char ' ', code 0x20, shift false
2020/10/23 16:40:39 packer-builder-qemu plugin: Sending char 'c', code 0x63, shift false
2020/10/23 16:40:39 packer-builder-qemu plugin: Sending char 'o', code 0x6F, shift false
2020/10/23 16:40:39 packer-builder-qemu plugin: Sending char 'n', code 0x6E, shift false
2020/10/23 16:40:40 packer-builder-qemu plugin: Sending char 's', code 0x73, shift false
2020/10/23 16:40:40 packer-builder-qemu plugin: Sending char 'o', code 0x6F, shift false
2020/10/23 16:40:40 packer-builder-qemu plugin: Sending char 'l', code 0x6C, shift false
2020/10/23 16:40:40 packer-builder-qemu plugin: Sending char 'e', code 0x65, shift false
2020/10/23 16:40:40 packer-builder-qemu plugin: Sending char '=', code 0x3D, shift false
2020/10/23 16:40:41 packer-builder-qemu plugin: Sending char 't', code 0x74, shift false
2020/10/23 16:40:41 packer-builder-qemu plugin: Sending char 't', code 0x74, shift false
2020/10/23 16:40:41 packer-builder-qemu plugin: Sending char 'y', code 0x79, shift false
2020/10/23 16:40:41 packer-builder-qemu plugin: Sending char 'S', code 0x53, shift true
2020/10/23 16:40:42 packer-builder-qemu plugin: Sending char '0', code 0x30, shift false
2020/10/23 16:40:42 packer-builder-qemu plugin: Sending char ' ', code 0x20, shift false
2020/10/23 16:40:42 packer-builder-qemu plugin: Sending char 'i', code 0x69, shift false
2020/10/23 16:40:42 packer-builder-qemu plugin: Sending char 'n', code 0x6E, shift false
2020/10/23 16:40:42 packer-builder-qemu plugin: Sending char 's', code 0x73, shift false
2020/10/23 16:40:43 packer-builder-qemu plugin: Sending char 't', code 0x74, shift false
2020/10/23 16:40:43 packer-builder-qemu plugin: Sending char '.', code 0x2E, shift false
2020/10/23 16:40:43 packer-builder-qemu plugin: Sending char 'c', code 0x63, shift false
2020/10/23 16:40:43 packer-builder-qemu plugin: Sending char 'm', code 0x6D, shift false
2020/10/23 16:40:43 packer-builder-qemu plugin: Sending char 'd', code 0x64, shift false
2020/10/23 16:40:44 packer-builder-qemu plugin: Sending char 'l', code 0x6C, shift false
2020/10/23 16:40:44 packer-builder-qemu plugin: Sending char 'i', code 0x69, shift false
2020/10/23 16:40:44 packer-builder-qemu plugin: Sending char 'n', code 0x6E, shift false
2020/10/23 16:40:44 packer-builder-qemu plugin: Sending char 'e', code 0x65, shift false
2020/10/23 16:40:44 packer-builder-qemu plugin: Special code '<enter>' found, replacing with: 0xFF0D
    qemu: No communicator is configured -- skipping StepWaitGuestAddress
==> qemu: Halting the virtual machine...
2020/10/23 16:40:45 packer-builder-qemu plugin: [INFO] communicator disabled, will not connect
2020/10/23 16:40:45 packer-builder-qemu plugin: Unable to load communicator config from state to populate provisionHookData
2020/10/23 16:40:45 packer-builder-qemu plugin: Running the provision hook
2020/10/23 16:40:45 packer-builder-qemu plugin: VM shut down.
==> qemu: Converting hard drive...
2020/10/23 16:40:45 packer-builder-qemu plugin: Executing qemu-img: []string{"convert", "-O", "qcow2", "output-qemu/packer-qemu", "output-qemu/packer-qemu.convert"}
2020/10/23 16:40:45 packer-builder-qemu plugin: stdout:
2020/10/23 16:40:45 packer-builder-qemu plugin: stderr: qemu-img: Could not open 'output-qemu/packer-qemu': Failed to get shared "write" lock
2020/10/23 16:40:45 packer-builder-qemu plugin: Is another process using the image [output-qemu/packer-qemu]?
2020/10/23 16:40:45 packer-builder-qemu plugin: Retryable error: QemuImg error: qemu-img: Could not open 'output-qemu/packer-qemu': Failed to get shared "write" lock
2020/10/23 16:40:45 packer-builder-qemu plugin: Is another process using the image [output-qemu/packer-qemu]?
==> qemu: Error getting file lock for conversion; retrying...
2020/10/23 16:40:46 packer-builder-qemu plugin: Executing qemu-img: []string{"convert", "-O", "qcow2", "output-qemu/packer-qemu", "output-qemu/packer-qemu.convert"}
2020/10/23 16:40:46 packer-builder-qemu plugin: stdout:
2020/10/23 16:40:46 packer-builder-qemu plugin: stderr:
2020/10/23 16:40:46 packer-builder-qemu plugin: failed to unlock port lockfile: close tcp 127.0.0.1:5914: use of closed network connection
2020/10/23 16:40:46 [INFO] (telemetry) ending qemu
2020/10/23 16:40:46 [INFO] (telemetry) Starting post-processor shell-local
==> qemu: Running post-processor: shell-local
2020/10/23 16:40:46 packer-post-processor-shell-local plugin: [INFO] (shell-local): Prepending inline script with #!/bin/bash -e
2020/10/23 16:40:46 packer-post-processor-shell-local plugin: [INFO] (shell-local): starting local command: /bin/sh -c PACKER_BUILDER_TYPE='qemu' PACKER_BUILD_NAME='qemu' PACKER_HTTP_ADDR='10.0.2.2:8621' PACKER_HTTP_IP='10.0.2.2' PACKER_HTTP_PORT='8621'  /tmp/packer-shell695533443
2020/10/23 16:40:46 packer-post-processor-shell-local plugin: [INFO] (shell-local communicator): Executing local shell command [/bin/sh -c PACKER_BUILDER_TYPE='qemu' PACKER_BUILD_NAME='qemu' PACKER_HTTP_ADDR='10.0.2.2:8621' PACKER_HTTP_IP='10.0.2.2' PACKER_HTTP_PORT='8621'  /tmp/packer-shell695533443]
==> qemu (shell-local): Running local shell script: /tmp/packer-shell695533443
    qemu (shell-local): Loading nbd...
    qemu (shell-local): Using /dev/nbd0
    qemu (shell-local): Binding image to /dev/nbd0...
    qemu (shell-local): /dev/nbd0 disconnected
    qemu (shell-local): Waiting for partitions to be created...
    qemu (shell-local): Mounting root partition...
==> qemu (shell-local): mount: /tmp/packer-maas-TJ4k: special device /dev/nbd0p1 does not exist.
2020/10/23 16:41:46 packer-post-processor-shell-local plugin: error: Script exited with non-zero exit status: 32.Allowed exit codes are: [0]
2020/10/23 16:41:46 [INFO] (telemetry) ending shell-local
2020/10/23 16:41:46 Deleting original artifact for build 'qemu'

* Post-processor failed: Script exited with non-zero exit status: 32.Allowed exit codes are: [0]
==> Wait completed after 1 minute 22 seconds
2020/10/23 16:41:46 machine readable: error-count []string{"1"}
==> Some builds didn't complete successfully and had errors:
2020/10/23 16:41:46 machine readable: qemu,error []string{"1 error(s) occurred:\n\n* Post-processor failed: Script exited with non-zero exit status: 32.Allowed exit codes are: [0]"}

* Post-processor failed: Script exited with non-zero exit status: 32.Allowed exit codes are: [0]
==> Builds finished but no artifacts were created.
Build 'qemu' errored after 1 minute 22 seconds: 1 error(s) occurred:
2020/10/23 16:41:46 [INFO] (telemetry) Finalizing.

* Post-processor failed: Script exited with non-zero exit status: 32.Allowed exit codes are: [0]

==> Wait completed after 1 minute 22 seconds

==> Some builds didn't complete successfully and had errors:
--> qemu: 1 error(s) occurred:

* Post-processor failed: Script exited with non-zero exit status: 32.Allowed exit codes are: [0]

==> Builds finished but no artifacts were created.
2020/10/23 16:41:46 waiting for all plugin processes to complete...
2020/10/23 16:41:46 /home/lee/go/bin/packer: plugin process exited
2020/10/23 16:41:46 /home/lee/go/bin/packer: plugin process exited
```

**Comment by tqre** [comment]
I get the same behavior with the "none" communicator.
The packer is speeding through the steps and prematurely halts the machine before installation has completed unless I use the pauses with debug mode as a workaround to wait until build scripts run their course.

So the

**Operating system and Environment details**

OS: Arch Linux
Arch: x86_64
Kernel: 5.9.1-arch1-1
qemu: 5.1.0
packer: 1.6.3

**Packer Buildfile**
```
build-debug.json

{
  "builders": [
    {
      "type": "qemu",
      "communicator": "none",
      "iso_url": "/home/lab/templates/iso/archlinux-2020.07.01-x86_64.iso",
      "iso_checksum": "md5:64aeb30b17cbfa74c2b4b17960623ad8",
      "output_directory": "VM",
      "vm_name": "build-debug.qcow2",
      "disk_size": "56G",
      "format": "qcow2",
      "memory": 4096,
      "cpus": 12,
      "http_directory": "files",
      "boot_wait": "1s",
      "boot_key_interval": "5ms",
      "shutdown_timeout": "120m",
      "boot_command": [
        "<enter><wait30s>",
        "curl http://{{ .HTTPIP }}:{{ .HTTPPort }}/files.tar --output files.tar<enter>",
        "tar xvf files.tar --strip-components=1<enter>",
        "chmod +x bootstrap.sh<enter>",
        "./bootstrap.sh<enter>",
        "poweroff<enter>"
      ]
    }
  ]
}
```

**Log Fragments and crash.log files**
```
packer build -debug build-debug.json
Debug mode enabled. Builds will not be parallelized.
qemu: output will be in this color.

==> qemu: Retrieving ISO
==> qemu: Trying /home/lab/templates/iso/archlinux-2020.07.01-x86_64.iso
==> qemu: Trying /home/lab/templates/iso/archlinux-2020.07.01-x86_64.iso?checksum=md5%3A64aeb30b17cbfa74c2b4b17960623ad8
==> qemu: /home/lab/templates/iso/archlinux-2020.07.01-x86_64.iso?checksum=md5%3A64aeb30b17cbfa74c2b4b17960623ad8 => /home/lab/templates/iso/archlinux-2020.07.01-x86_64.iso
==> qemu: Pausing after run of step 'StepDownload'. Press enter to continue.
==> qemu: Pausing after run of step 'stepPrepareOutputDir'. Press enter to continue.
==> qemu: Pausing after run of step 'StepCreateFloppy'. Press enter to continue.
==> qemu: Pausing after run of step 'StepCreateCD'. Press enter to continue.
==> qemu: Creating required virtual machine disks
==> qemu: Pausing after run of step 'stepCreateDisk'. Press enter to continue.
==> qemu: Pausing after run of step 'stepCopyDisk'. Press enter to continue.
==> qemu: Pausing after run of step 'stepResizeDisk'. Press enter to continue.
==> qemu: Pausing after run of step 'stepHTTPIPDiscover'. Press enter to continue.
==> qemu: Starting HTTP server on port 8325
==> qemu: Pausing after run of step 'StepHTTPServer'. Press enter to continue.
    qemu: No communicator is set; skipping port forwarding setup.
==> qemu: Pausing after run of step 'stepPortForward'. Press enter to continue.
==> qemu: Looking for available port between 5900 and 6000 on 127.0.0.1
==> qemu: Pausing after run of step 'stepConfigureVNC'. Press enter to continue.
==> qemu: Starting VM, booting from CD-ROM
==> qemu: Pausing after run of step 'stepRun'. Press enter to continue.
==> qemu: Pausing after run of step 'stepConfigureQMP'. Press enter to continue.
==> qemu: Waiting 1s for boot...
==> qemu: Connecting to VM via VNC (127.0.0.1:5918)
==> qemu: Typing the boot command over VNC...
==> qemu: Pausing after run of step 'boot_command: <enter><wait30s>curl http://10.0.2.2:8325/files.tar --output files.tar<enter>tar xvf files.tar --strip-components=1<enter>chmod +x bootstrap.sh<enter>./bootstrap.sh<enter>#poweroff<enter>'. Press enter to continue.
==> qemu: Pausing after run of step 'stepTypeBootCommand'. Press enter to continue.
    qemu: No communicator is configured -- skipping StepWaitGuestAddress
==> qemu: Pausing after run of step 'stepWaitGuestAddress'. Press enter to continue.
==> qemu: Pausing after run of step 'StepConnect'. Press enter to continue.
==> qemu: Pausing after run of step 'StepProvision'. Press enter to continue.
==> qemu: Pausing after run of step 'StepCleanupTempKeys'. Press enter to continue.
==> qemu: Halting the virtual machine...
```

**Comment by SwampDragons** [comment]
Found it. PR is linked, test binaries are here: 3577c4a2831e44f6335390eb65b27301bdc09b6f

Sorry about that, friends.
